enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,225 @@
+---
+title: optional-ingress
+authors:
+  - "@alebedev87"
+reviewers:
+  - "@Miciah"
+  - "@bparees"
+  - "@cewong"
+  - "@wking"
+approvers:
+  - "@Miciah"
+  - "@bparees"
+  - "@wking"
+api-approvers:
+  - N/A
+creation-date: 2023-06-02
+last-updated: 2023-06-02
+tracking-link:
+  - https://issues.redhat.com/browse/NE-1129
+see-also:
+  - "/enhancements/installer/component-selection.md"
+  - "/enhancements/ingress/aws-load-balancer-operator.md"
+replaces:
+superseded-by:
+---
+
+# Make Ingress Operator optional
+
+## Summary
+
+This enhancement describes the implementation of a new capability to enable/disable the default ingress of the OpenShift cluster.
+The new capability (from here on called the ingress capability) includes [the cluster ingress operator](https://github.com/openshift/cluster-ingress-operator) as well as the custom resources managed by the operator.
+This enhancement focuses on the requirements for [RFE-3395](https://issues.redhat.com/browse/RFE-3395) which targets ROSA, a managed OpenShift service based on HyperShift. The implementation of the ingress capability on the standalone OpenShift is out of the scope of this enhancement.
+
+## Motivation
+
+Users of the ROSA managed OpenShift service would like to use [AWS load balancers](/enhancements/ingress/aws-load-balancer-operator.md) as a way to accept the ingress traffic into the cluster.
+The cluster ingress operator becomes unnecessary and can be disabled to reduce the resource consumption.
+
+### Goals
+- Implement the new ingress capability.
+- Add the ingress capability to the ingress operator's payload.
+- Make the dependent cluster operators tolerate the absence of the ingress operator.
+
+### Non-Goals
+- Use a different way for enabling/disabling the ingress other than via [the cluster capability](/enhancements/installer/component-selection.md).
+- Fully implement the ingress capability for the standalone OpenShift:
+  - Make the OpenShift installer tolerate the absence of the default ingress controller.
+  - Make the cluster authentication operator tolerate the absence of the default ingress controller.
+  - Make the cluster monitoring operator tolerate the absence of the default ingress controller.
+- Disable the Route API or try to tolerate the absence of the default ingress controller.
+- Implement the ingress operator disabling in the HyperShift's control plane operator.
+
+## Proposal
+
+### User Stories
+
+- As a HyperShift engineer, I want to be able to disable the ingress operator on the hosted control plane.
+- As an OpenShift engineer, I want the cluster operators used by HyperShift to tolerate the absence of the ingress operator.
```

Comment:
why does an openshift engineer want that?

Reply:
Well, I think it's in the interest of many actors but I agree the openshift engineer may be a little less interested that the hypershift engineer. I changed to the latter.